alf/algorithms/mbrl_algorithm.py

```diff
@@ -174,8 +174,8 @@ def _calc_step_reward(self, obs, action, state: MbrlState):
     def _predict_with_planning(self, time_step: TimeStep, state,
                                epsilon_greedy):
         # full state in
-        action = self._planner_module.generate_plan(time_step, state,
-                                                    epsilon_greedy)
+        action, state = self._planner_module.generate_plan(
+            time_step, state, epsilon_greedy)
```

Comment:
pass in  `state.planner` instead as planner_module should not know about the structure of `state` 
and generate_plan => predict_plan

Reply:
I agree that planner_module should not know about the structure of state. Currently there are still some flaws in the interface design.

Reply:
Now changed name to predict_plan